client.go

```diff
@@ -66,6 +71,25 @@ type Client struct {
 	Options []ClientOption
 }
 
+func (c *Client) umask() os.FileMode {
+	if c == nil {
+		um := syscall.Umask(0)
+		syscall.Umask(um)
```

Comment:
> Why are we setting the value we just received

Sadly, this is the most portable way to read umask of process:

> It is impossible to use umask() to fetch a process's umask without at
> the same time changing it.  A second call to umask() would then be
> needed to restore the umask.  The nonatomicity of these two steps
> provides the potential for races in multithreaded programs.
>
> Since Linux 4.7, the umask of any process can be viewed via the Umask
> field of /proc/[pid]/status.  Inspecting this field in
> /proc/self/status allows a process to retrieve its umask without at
> the same time changing it.

http://man7.org/linux/man-pages/man2/umask.2.html